### PR TITLE
♻️ [RUM-5294] Use performanceObserver for navigation entries

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Following the introduction of `performanceObservable` to move away from `performanceCollection`: [PR](https://github.com/DataDog/browser-sdk/pull/2818)
Let use `performanceObservable` for navigation entries.

## Changes
- Remove navigation timings logic from `performanceCollection`
- Use `perfomanceObservable` in `trackViewTimings`

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts [hunk 1]
```diff
@@ -1,30 +1,71 @@
...
+    ;({ notifyPerformanceEntries } = mockPerformanceObserver())
+    stop = trackNavigationTimings({} as RumConfiguration, navigationTimingsCallback).stop
+    notifyPerformanceEntries([
+      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, { loadEventEnd: 0 as RelativeTime }),
     ])
 
+    expect(navigationTimingsCallback).not.toHaveBeenCalled()
+  })
+
+  it('should provide navigation timing when navigation timing is not supported ', () => {
```
Comment (amortemousque) [inline packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts:59]: Add a test for the fallback strategy

File: packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts
Comment (amortemousque) [inline packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts:1]: Improved our test coverage by adding tests:
- should discard incomplete navigation timing
- should provide navigation timing when navigation timing is not supported

File: packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts [hunk 2]
```diff
@@ -1,30 +1,71 @@
...
+      window.PerformanceObserver = originalPerformanceObserver
+    })
+  }
 
   beforeEach(() => {
     navigationTimingsCallback = jasmine.createSpy()
+  })
 
-    setupBuilder = setup().beforeBuild(({ lifeCycle }) => trackNavigationTimings(lifeCycle, navigationTimingsCallback))
+  afterEach(() => {
```
Comment (nulrich) [inline packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts:30]: **🥜 nitpick:** ‏use registerCleanupTask

File: packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts [hunk 3]
```diff
@@ -1,30 +1,71 @@
...
+      domInteractive: 234 as Duration,
+      loadEvent: 567 as Duration,
+    })
+  })
 
-    lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION),
+  it('should discard incomplete navigation timing', () => {
+    ;({ notifyPerformanceEntries } = mockPerformanceObserver())
+    stop = trackNavigationTimings({} as RumConfiguration, navigationTimingsCallback).stop
```
Comment (nulrich) [inline packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts:51]: ‏Why not using the same syntax as above? `;({ stop } = trackNavigationTimings({} as RumConfiguration, navigationTimingsCallback))`

File: packages/rum-core/src/domain/longTask/longTaskCollection.spec.ts [hunk 4]
```diff
@@ -22,10 +22,12 @@ describe('long task collection', () => {
   })
 
   it('should only listen to long task performance entry', () => {
+    const { notifyPerformanceEntries } = mockPerformanceObserver()
     const { lifeCycle, rawRumEvents } = setupBuilder.build()
+
+    notifyPerformanceEntries([createPerformanceEntry(RumPerformanceEntryType.NAVIGATION)])
```
Comment (nulrich) [inline packages/rum-core/src/domain/longTask/longTaskCollection.spec.ts:28]: **❓ question:** ‏Why do we need to notify the `RumPerformanceEntryType.NAVIGATION`? longTaskCollection only listen to lifeCycle.

File: packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts (hunk 2 above)
Comment (N-Boutaib) [reply packages/rum-core/src/domain/view/viewMetrics/trackNavigationTimings.spec.ts:30]: You can also remove `clock?.cleanup()` if you're up to date with main

**Review by N-Boutaib — APPROVED** [review]

File: packages/rum-core/src/domain/longTask/longTaskCollection.spec.ts (hunk 4 above)
Comment (amortemousque) [reply packages/rum-core/src/domain/longTask/longTaskCollection.spec.ts:28]: Once the refacto will be completed, longTask will use PerformanceObservable. So ultimately this test will ensure that PerformanceObservable is well configured

**Review by nulrich — APPROVED** [review]

**Review by N-Boutaib — APPROVED** [review]

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**